core/capabilities/ccip/ccipevm/extradatadecoder.go

```diff
@@ -0,0 +1,56 @@
+package ccipevm
+
+import (
+	"fmt"
+
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+)
+
+func DecodeDestExecDataToMap(DestExecData cciptypes.Bytes) (map[string]interface{}, error) {
+	destGasAmount, err := abiDecodeUint32(DestExecData)
+	if err != nil {
+		return nil, fmt.Errorf("decode dest gas amount: %w", err)
+	}
+
+	return map[string]interface{}{
+		evmDestExecDataKey: destGasAmount,
+	}, nil
+}
+
+func DecodeExtraArgsToMap(extraArgs []byte) (map[string]any, error) {
+	if len(extraArgs) < 4 {
+		return nil, fmt.Errorf("extra args too short: %d, should be at least 4 (i.e the extraArgs tag)", len(extraArgs))
+	}
+
+	var method string
+	var extraByteOffset int
+	switch string(extraArgs[:4]) {
+	case string(evmExtraArgsV1Tag):
+		// for EVMExtraArgs, the first four bytes is the method name
+		method = evmV1DecodeName
+		extraByteOffset = 4
+	case string(evmExtraArgsV2Tag):
+		method = evmV2DecodeName
+		extraByteOffset = 4
+	case string(svmExtraArgsV1Tag):
+		// for SVMExtraArgs there's the four bytes plus another 32 bytes padding for the dynamic array
+		// https://github.com/smartcontractkit/chainlink/blob/33c0bda696b0ed97f587a46eacd5c65bed9fb2c1/contracts/src/v0.8/ccip/libraries/Client.sol#L57
+		// this is a temporary solution, the evm on-chain side will fix it, so the offset should just be 4 instead of 36
```

Comment:
Thanks for the link, but this doesn't really answer why there are 32 bytes of padding. I'm imagining that this behavior could change in the future, maybe we add a new SVM arg or there is an EVM optimization. When that causes the "abi decode extra args" error below, we should give the future dev something more concrete to work with.

Reply:
I've created an unassigned ticket https://smartcontract-it.atlassian.net/browse/BCI-4180